### ISSUE TITLE
celeba aligned and wild

### ISSUE THREAD

**Issue by x2x5** [ubc-vision/StableKeypoints#10]
Hi, I just don't understand the directory structure of the celeba dataset.

I saw your mentioned script in

https://github.com/xingzhehe/GANSeg/tree/main/data/celeba_wild_raw/MAFL

Then I don't understand whether you use same python file for `celeba_wild_preprocess.py` for wild and aligned celeba data?

because I only saw "wild preprocess" python file.

**Comment by ehedlin** [comment]
The aligned images are cropped and centred acording to the bounding boxes (aligning and cropping is already performed and available on the celeba website) and in the wild are the subset of unaligned images whose faces take up more than 30% of the image. [celeba_wild_preprocess.py](https://github.com/xingzhehe/AutoLink-Self-supervised-Learning-of-Human-Skeletons-and-Object-Outlines-by-Linking-Keypoints/blob/main/datasets/preprocess/celeba_wild_preprocess.py) finds the in the wild images by filtering with that 30% threshold. This same logic is handled in [this line](https://github.com/ubc-vision/StableKeypoints/blob/12b8f92d672d59f1c10ea2dbc537b650ccdd8a41/datasets/celeba.py#L52) so preprocessing isnt necessary.

**Comment by ehedlin** [comment]
Its also worth noting that all aligned images are used so the size of that set will be larger than unaligned.

**Comment by tian-2024** [comment]
> Its also worth noting that all aligned images are used so the size of that set will be larger than unaligned.

Thank you so much!

so is the Celeba dataset directory looks like this?
![image](https://github.com/user-attachments/assets/25adcbb8-dfe3-4932-86ba-8ecf8db12324)

**Comment by x2x5** [comment]
> The aligned images are cropped and centred acording to the bounding boxes (aligning and cropping is already performed and available on the celeba website) and in the wild are the subset of unaligned images whose faces take up more than 30% of the image. [celeba_wild_preprocess.py](https://github.com/xingzhehe/AutoLink-Self-supervised-Learning-of-Human-Skeletons-and-Object-Outlines-by-Linking-Keypoints/blob/main/datasets/preprocess/celeba_wild_preprocess.py) finds the in the wild images by filtering with that 30% threshold. This same logic is handled in [this line](https://github.com/ubc-vision/StableKeypoints/blob/12b8f92d672d59f1c10ea2dbc537b650ccdd8a41/datasets/celeba.py#L52) so preprocessing isnt necessary.

I read the code more deeply, so now I can understand some parts of what you said, thanks a lot!

so do you mean when using celeba datasets, I just need to unzip all data and put MAFL directory in CelebA, and I don't need other preprocessing?

like this, right?
![image](https://github.com/user-attachments/assets/8d74c9d7-99db-49ec-920d-a475d4006279)

**Comment by x2x5** [comment]
I also wonder some questions about the CUB dataset.

I downloaded the raw dataset, it's 1.1 GB.
```
1.1G 9月   3 19:01 CUB_200_2011.tgz
```

but when I used the preprocessing python file and get 'cub.h5', it's just 555MB.

```
555M 9月   3 19:34 cub.h5
```

so I wonder if it's the same as the raw dataset?

because I never use the 1.1GB dataset to get 'cub.h5'.

I just use the cachedir file.

![image](https://github.com/user-attachments/assets/05a6b703-ada6-4d3d-86f8-770054ce346b)

it's just 27.6 MB

![image](https://github.com/user-attachments/assets/fd7b97cf-7566-4141-9a51-3ca8b6c9fabb)

I don't undetstand this part.

so I guess the cub.h5 is a cleaned version of the raw data and we just don't need to use the raw data, right?

Thank you so much~
